js/serial.js

```diff
@@ -12,10 +15,76 @@ var serial = {
     transmitting:   false,
     outputBuffer:  [],
 
+    LOGHEAD: 'SERIAL: ',
+
     connect: function (path, options, callback) {
         var self = this;
         self.openRequested = true;
 
+        var testUrl = path.match(/tcp:\/\/(.*):(.*)/)
+        if (testUrl) {
+            self.connectionIP = testUrl[1];
+            self.connectionPort = testUrl[2] || self.connectionPort;
```

Comment:
`testUrl[2]` will always be defined. If you want it to default to the predefined port, make the port number optional in the regex: `/^http:\/\/([A-Za-z0-9\.-]+)(?:\:(\d+))?$`.